### PR TITLE
Math formatting and old code updates

### PR DESCRIPTION
**Pull request by LightnessOfBeing**
Added proper Latex formatting. Some outdated code refactored mainly using  Java 8 features.

### CODE REVIEW

**Comment by AlexandruValeanu** [comment]
What's the point of using LaTeX in regular comments? It just makes them more difficult to follow.

**Comment by jkinable** [comment]
Thanks for your submission. Our javadoc will look very nice and shiny. 

Following @AlexandruValeanu : can you remove the Latex formatting from the inline comments? Only use latex formatting in je Javadoc. After all, only the javadoc gets parsed by the latex parser. There is no way to have latex formatted inline comments.

**Comment by Toptachamann** [comment]
I agree with the point on harder readability. This problem could be solved with some plugin for instant javadoc compilation in Intellij. I've searched for something similar and found only plugins for .tex  file editing. Maybe you know how Javadoc with latex pieces could be compilated instantly? Currently, the documentation produced with `Ctrl+Q` doesn't compiles latex.

**Comment by jkinable** [comment]
@Toptachamann I think you misunderstood. The latex in the javadoc is great. There's no need to have instant compilation in Intellij (or any of the other IDEs). Javadoc is automatically generated as part of our release process. You can simulate this by running:

`mvg javadoc:aggregate`

If you then open any of the generated javadoc pages in your webbrowser, you will see the parsed latex. It should look like this: http://jgrapht.org/javadoc/org/jgrapht/alg/scoring/ClosenessCentrality.html

For the inline comments (so not the javadoc): these are just for coding purposes. Inline comments are never part of our published documentation. Consequently these don't require latex markup.

**Comment by Toptachamann** [comment]
Oops, yes. I didn't thought about generated Javadoc pages, so they solve my problem

File: jgrapht-core/src/main/java/org/jgrapht/alg/cycle/JohnsonSimpleCycles.java [hunk 1]
```diff
@@ -151,9 +151,9 @@ public void setGraph(Graph<V, E> graph)
     private Pair<Graph<V, E>, Integer> findMinSCSG(int startIndex)
     {
         /*
-         * Per Johnson : "adjacency structure of strong component K with least vertex in subgraph of
-         * G induced by {s, s+ 1, n}". Or in contemporary terms: the strongly connected component of
-         * the subgraph induced by {v1,...,vn} which contains the minimum (among those SCCs) vertex
+         * Per Johnson : "adjacency structure of strong component $K$ with least vertex in subgraph of
+         * $G$ induced by $(s, s + 1, n$". Or in contemporary terms: the strongly connected component of
```
Comment (jkinable) [inline jgrapht-core/src/main/java/org/jgrapht/alg/cycle/JohnsonSimpleCycles.java]: Missing parenthesis ')'

File: jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/KShortestPaths.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
  *
  * <p>
  * The algorithm is a variant of the Bellman-Ford algorithm but instead of only storing the best
- * path it stores the "k" best paths at each pass, yielding a complexity of O(k*n*(m^2)) where m is
+ * path it stores the "k" best paths at each pass, yielding a complexity of $O(k \cdot n \cdot (m^2))$ where m is
```
Comment (jkinable) [inline jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/KShortestPaths.java]: $m$ and $n$

File: jgrapht-core/src/main/java/org/jgrapht/alg/vertexcover/GreedyVCImpl.java [hunk 3]
```diff
@@ -82,12 +82,11 @@
             ux.addNeighbor(vx);
             vx.addNeighbor(ux);
 
-            assert (ux.neighbors.get(vx) == vx.neighbors.get(
-                ux)) : " in an undirected graph, if vx is a neighbor of ux, then ux must be a neighbor of vx";
+            assert (Objects.equals(ux.neighbors.get(vx), vx.neighbors.get(
```
Comment (jkinable) [inline jgrapht-core/src/main/java/org/jgrapht/alg/vertexcover/GreedyVCImpl.java]: This should be ==, not equals. We want to compare the objects by reference.

File: jgrapht-core/src/main/java/org/jgrapht/alg/flow/PushRelabelMFImpl.java [hunk 4]
```diff
@@ -284,17 +284,17 @@ private void gapHeuristic(int l){
...
-        downhill from $u$.
+        We can relabel an overflowing vertex $u$ if for every vertex v for which there is
+        residual capacity from u to v, flow cannot be pushed from u to v because v is not
+        downhill from u.
      */
     private void relabel(VertexExtension ux){
         int oldHeight = ux.height;
 
-        // Increase the height of $u$; $u.h = 1 + \min(v.h : (u, v) \in E_f}$
+        // Increase the height of u; u.h = 1 + min(v.h : (u, v) \in E_f)
```
Comment (AlexandruValeanu) [inline jgrapht-core/src/main/java/org/jgrapht/alg/flow/PushRelabelMFImpl.java]: Change \in to in and E_f to Ef

File: jgrapht-core/src/main/java/org/jgrapht/util/FibonacciHeap.java [hunk 5]
```diff
@@ -22,8 +22,8 @@
 /**
  * This class implements a Fibonacci heap data structure. Much of the code in this class is based on
  * the algorithms in the "Introduction to Algorithms" by Cormen, Leiserson, and Rivest in Chapter
- * 21. The amortized running time of most of these methods is O(1), making it a very fast data
- * structure. Several have an actual running time of O(1). removeMin() and delete() have O(log n)
+ * 21. The amortized running time of most of these methods is $O(1)$, making it a very fast data
+ * structure. Several have an actual running time of $O(1)$. removeMin() and delete() have O(log n)
```
Comment (jkinable) [inline jgrapht-core/src/main/java/org/jgrapht/util/FibonacciHeap.java]: $O(\log n)$

**Comment by jkinable** [comment]
Very nice, thanks. This brings us one step closer to nice, consistent javadoc. Many classes still have poorly structured javadoc so there's still work to be done.

**Pull request merged**